docs: Add multichain Safe deployments guide (#617)
safe-global/safe-docs: pages/sdk/protocol-kit/guides/safe-deployment/deploy-safe.mdx

import { Steps } from 'nextra/components'

# Deploy a Safe

In this guide, you will learn how to deploy a new Safe using the Protocol Kit. This process includes initializing the Protocol Kit, setting up your Safe configuration, and executing the deployment.

For more detailed information, see the [Protocol Kit Reference](../../../protocol-kit/reference/safe.mdx).

## Prerequisites

- [Node.js and npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)

## Install dependencies

First, you need to install the Protocol Kit.

```bash
pnpm add @safe-global/protocol-kit
```

## Steps

<Steps>

  ### Imports

>   Begin by importing the `Safe` class from the Protocol Kit.

Reviewer: Perhaps using "begin" here and in the second step "firstly" is not super clean

Author: done!

  {/* <!-- vale off --> */}

  ```typescript
  import Safe from '@safe-global/protocol-kit'
  ```

  {/* <!-- vale on --> */}

  ### Create a signer

>   Firstly, you need to get a signer, this example uses a private key, but any way to get an `EIP-1193` compatible signer can be used.

Reviewer: This is not super accurate. Something like this could be better? :

"You need a signer to instantiate the protocol-kit. In the following example a private key is being used but you can specify as well an address. More about the available options in the [protocol-kit reference](https://safe_deployment_guide_v2--docs.review.5afe.dev/sdk/protocol-kit/reference/safe#init)"

Author: done!

  {/* <!-- vale off --> */}

  ```typescript
  const SIGNER_PRIVATE_KEY = // ...
  const RPC_URL = 'https://rpc.ankr.com/eth_sepolia'
  ```

  {/* <!-- vale on --> */}

>   ### Initialize the `ProtocolKit`

Reviewer: ```suggestion
  ### Initialize the `protocol-kit`
```

Reviewer 2: ```suggestion
  ### Initialize the Protocol Kit
```

When writing about the Protocol Kit we are using the form of full name. The kebap case is only used for imports and code snippets.

  {/* <!-- vale off --> */}

>   Initialize the Protocol Kit with the `provider`, `signer`, and the `predictedSafe` configuration.

Reviewer: I will ellaborate a little more here as well. Why we use the predictedSafe prop for example and link to the configuration in the reference guide

Author: done!

  ```typescript
  // Define your Safe configuration
  const predictedSafe = {
    safeAccountConfig: {
      owners: ['0xOwner1', '0xOwner2', '0xOwner3'], // In this example, 3 owners
      threshold: 2 // Requires at least 2 signatures to approve a transaction
    }
  }

  // Initialize the Protocol Kit with the predictedSafe
  const protocolKit = await Safe.init({
    provider: RPC_URL,
    signer: SIGNER_PRIVATE_KEY,
    predictedSafe
  })
  ```

  {/* <!-- vale on --> */}

  ### Predict your Safe address

>   you can predict your Safe address using the `getAddress` method in the Protocol Kit.

Reviewer: ```suggestion
  You can predict your Safe address by using the `getAddress` method in the Protocol Kit.
```

  {/* <!-- vale off --> */}

  ```typescript
  const safeAddress = await protocolKit.getAddress()
  ```

  {/* <!-- vale on --> */}

>   ### Create the deployement transaction

Reviewer: ```suggestion
  ### Create the deployment transaction
```

>   Now, generate the transaction to deploy the Safe and execute it using the Protocol Kit.

Reviewer: ```suggestion
  In this step we are going to create and execute the transaction to deploy a new Safe account.
```

  {/* <!-- vale off --> */}

  ```typescript
  const deploymentTransaction = await protocolKit.createSafeDeploymentTransaction()
  ```

  {/* <!-- vale on --> */}

>   ### Execute the deployement transaction

Reviewer: ```suggestion
  ### Execute the deployment transaction
```

  You can execute the deployement transaction using the integrated signer or your preferred external Ethereum client.

  {/* <!-- vale off --> */}

  ```typescript
  const client = await protocolKit.getSafeProvider().getExternalSigner()

  const txHash = await client.sendTransaction({
    to: deploymentTransaction.to,
    value: BigInt(deploymentTransaction.value),
    data: deploymentTransaction.data as `0x${string}`,
    chain: sepolia
  })

  const txReceipt = await client.waitForTransactionReceipt({ hash: txHash })
  ```

>   ### Reconnect the `protocolKit` to the newly deployed Safe

Reviewer 2: ```suggestion
  ### Reconnect the Protocol Kit to the newly deployed Safe
```
